Web Inspector: hard to disable inline breakpoints without entirely removing them

dcrousso opened the pull request:
#### 6ccba2787ef21faafedfda05a69e84064f1a07e8
<pre>
Web Inspector: hard to disable inline breakpoints without entirely removing them
<a href="https://bugs.webkit.org/show_bug.cgi?id=246234">https://bugs.webkit.org/show_bug.cgi?id=246234</a>

Reviewed by Patrick Angle.

This better matches the user expectation that clicking will toggle the disabled state, not remove
the (inline) breakpoint. Removing the (inline) breakpoint can be done via context menu, dragging the
breakpoint icon out of the gutter are for that line, or pressing delete when that breakpoint is
selected in the Breakpoints section of the navigation sidebar in the Sources Tab.

In order to differentiate a placeholder inline breakpoint vs an existing breakpoint, the former is
rendered as the outline of a breakpoint icon, whereas the latter is a solid color (unless it&apos;s set
to auto-continue, in which case it will have a small carve-out in the arrow).

* Source/WebInspectorUI/UserInterface/Views/BreakpointInlineWidget.js:
(WI.BreakpointInlineWidget):
(WI.BreakpointInlineWidget.prototype._update):
(WI.BreakpointInlineWidget.prototype._handleClick):
(WI.BreakpointInlineWidget.prototype._handleBreakpointsEnabledDidChange): Added.
* Source/WebInspectorUI/UserInterface/Views/BreakpointInlineWidget.css:
(.inline-widget.breakpoint):
(.inline-widget.breakpoint.placeholder::before): Added.
(.inline-widget.breakpoint:not(.resolved)): Added.
(.inline-widget.breakpoint:not(.placeholder, .disabled)): Renamed from `.inline-widget.breakpoint:not(.disabled)`.
(.inline-widget.breakpoint.auto-continue::after):
Drive-by: Also handle when breakpoints are globally disabled.

Canonical link: <a href="https://commits.webkit.org/255292@main">https://commits.webkit.org/255292@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fde8ad35ccfac9c77c2e6f03d9117b6a03770560

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/92055 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/1291 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/22578 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/101856 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/161915 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/1287 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/29698 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/84507 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/98037 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/97713 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/803 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/78588 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/84507 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/82725 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/22578 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/84507 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/66/builds/36124 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/22578 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/67/builds/33871 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/22578 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/63/builds/37743 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/78588 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1656 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/39634 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/22578 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

patrickangle reviewed (approved):
Nice - thanks for the screenshot too 😅

dcrousso committed:
Web Inspector: hard to disable inline breakpoints without entirely removing them
https://bugs.webkit.org/show_bug.cgi?id=246234

Reviewed by Patrick Angle.

This better matches the user expectation that clicking will toggle the disabled state, not remove
the (inline) breakpoint. Removing the (inline) breakpoint can be done via context menu, dragging the
breakpoint icon out of the gutter are for that line, or pressing delete when that breakpoint is
selected in the Breakpoints section of the navigation sidebar in the Sources Tab.

In order to differentiate a placeholder inline breakpoint vs an existing breakpoint, the former is
rendered as the outline of a breakpoint icon, whereas the latter is a solid color (unless it's set
to auto-continue, in which case it will have a small carve-out in the arrow).

* Source/WebInspectorUI/UserInterface/Views/BreakpointInlineWidget.js:
(WI.BreakpointInlineWidget):
(WI.BreakpointInlineWidget.prototype._update):
(WI.BreakpointInlineWidget.prototype._handleClick):
(WI.BreakpointInlineWidget.prototype._handleBreakpointsEnabledDidChange): Added.
* Source/WebInspectorUI/UserInterface/Views/BreakpointInlineWidget.css:
(.inline-widget.breakpoint):
(.inline-widget.breakpoint.placeholder::before): Added.
(.inline-widget.breakpoint:not(.resolved)): Added.
(.inline-widget.breakpoint:not(.placeholder, .disabled)): Renamed from `.inline-widget.breakpoint:not(.disabled)`.
(.inline-widget.breakpoint.auto-continue::after):
Drive-by: Also handle when breakpoints are globally disabled.

Canonical link: https://commits.webkit.org/255292@main
